tests/slurm_plugin/test_resume.py

```diff
@@ -367,17 +376,289 @@ def test_resume_config(config_file, expected_attributes, test_datadir, mocker):
+        (
+            [
+                SimpleNamespace(name="queue1-dy-c5xlarge-1", state_string="ALLOCATED+CLOUD+NOT_RESPONDING+POWERING_UP"),
+                SimpleNamespace(name="queue1-dy-c5xlarge-2", state_string="ALLOCATED+CLOUD+NOT_RESPONDING+POWERING_UP"),
+                SimpleNamespace(name="queue1-st-c5xlarge-1", state_string="ALLOCATED+CLOUD+NOT_RESPONDING+POWERING_UP"),
+                SimpleNamespace(name="queue1-st-c5xlarge-2", state_string="ALLOCATED+CLOUD+NOT_RESPONDING+POWERING_UP"),
+            ],
+            3,
+            False,
+            [
+                {
+                    "Instances": [
+                        {
+                            "InstanceId": "i-11111",
+                            "InstanceType": "c5.xlarge",
+                            "PrivateIpAddress": "ip.1.0.0.1",
+                            "PrivateDnsName": "ip-1-0-0-1",
+                            "LaunchTime": datetime(2020, 1, 1, tzinfo=timezone.utc),
+                            "NetworkInterfaces": [
+                                {
+                                    "Attachment": {
+                                        "DeviceIndex": 0,
+                                        "NetworkCardIndex": 0,
+                                    },
+                                    "PrivateIpAddress": "ip.1.0.0.1",
+                                },
+                            ],
+                        },
+                    ]
+                },
+                client_error("InsufficientReservedInstanceCapacity"),
+            ],
+            {"InsufficientReservedInstanceCapacity": {"queue1-st-c5xlarge-2"}},
+            [call(["queue1-dy-c5xlarge-1"], nodeaddrs=["ip.1.0.0.1"], nodehostnames=None)],
+            dict(
+                zip(
+                    ["queue1-dy-c5xlarge-1"],
+                    [
+                        EC2Instance("i-11111", "ip.1.0.0.1", "ip-1-0-0-1", datetime(2020, 1, 1, tzinfo=timezone.utc)),
+                    ],
+                )
+            ),
+            True,
+            True,
         ),
     ],
     ids=[
-        "all_or_nothing without ICE error",
-        "all_or_nothing with ICE error",
-        "best_effort without ICE error",
-        "best_effort with ICE error",
+        "node list scaling + all_or_nothing without ICE error",
+        "node list scaling + all_or_nothing with ICE error",
+        "node list scaling + best_effort without ICE error",
+        "node list scaling + best_effort with ICE error",
         "invalid_heartbeat",
+        "job level scaling + empty resume file + all_or_nothing without ICE error",
+        "job level scaling + empty resume file + all_or_nothing with ICE error",
+        "job level scaling + empty resume file + best_effort without ICE error",
+        "job level scaling + empty resume file + best_effort with ICE error",
     ],
 )
-def test_resume_launch(
+def test_resume_node_list_scaling(
```

Comment:
This test covers both node-list and job-level scaling. Could we keep the name as `test_resume_launch` or `test_resume_scaling`?